### PR TITLE
refactor(types): port integration types etc to explicit imports

### PR DESCRIPTION
**Pull request by tjgupta**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @tjgupta has signed the CLA. Thanks for the pull request!

File: src/common/types/integrations.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// @flow
+import { FormattedMessage } from 'react-intl';
+import { APP_INTEGRATION } from '../../constants';
+
+type AppIntegrationAPIMiniItem = {
+    id: string,
+    type: typeof APP_INTEGRATION,
```
Comment (tjgupta) [inline src/common/types/integrations.js]: This `typeof` is the only real change in this PR (besides explicit imports). Same thing on line 52, plus a corresponding change int the constants file.

File: src/common/types/integrations.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+};
+
+type ExecuteAPIParam = {
+    key: string,
+    value: string,
+};
+
+type ExecuteAPI = {
+    integration_type: string,
+    method: HTTP_POST | HTTP_GET,
```
Comment (alexkrolick) [inline src/common/types/integrations.js]: these might need a `typeof`?

Comment (tjgupta) [reply src/common/types/integrations.js]: Good catch, done @alexkrolick

File: src/common/types/integrations.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+};
+
+type ExecuteAPIParam = {
+    key: string,
+    value: string,
+};
+
+type ExecuteAPI = {
+    integration_type: string,
+    method: typeof HTTP_POST | typeof HTTP_GET,
```
Comment (tjgupta) [inline src/common/types/integrations.js]: Actually, added `typeof` here too after @alexkrolick mentioned it.

**Pull request merged**